### PR TITLE
Type definitions for @materializecss/materialize (materializecss__materialize)

### PR DESCRIPTION
**Pull request by mauromascarenhas**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## materializecss__materialize ([<kbd>unpkg</kbd>](https://unpkg.com/browse/materializecss__materialize@latest/))
was missing the following properties:
1. version
1. keys
1. tabPressed
1. keyDown
1. initializeJqueryWrapper

<details>
<summary>as well as these 10 other properties...</summary>
<p>objectSelectorString, guid, escapeHash, checkWithinContainer, checkPossibleAlignments, getOverflowParent, getIdFromTrigger, getDocumentScrollTop, getDocumentScrollLeft, throttle</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 6807750541a042b753872a67b15998186e30ef52
</p>

**Comment by peterblazejewicz** [comment]
>  The package does not already provide its own types, or cannot have its .d.ts files generated via --declaration

so what's the relation to v1.0 of the types?
https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/materialize-css

**Comment by mauromascarenhas** [comment]
> > The package does not already provide its own types, or cannot have its .d.ts files generated via --declaration
> 
> so what's the relation to v1.0 of the types? https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/materialize-css

[@materializecss/materialize](https://github.com/materializecss/materialize/) is a community-driven project forked from the "original [materialize-css](https://github.com/Dogfalo/materialize/)", which is targeted by "@types/materialize-css". The declarations are almost the same, but the latest release (v1.1) has some new attributes and fixes for [XSS vulnerabilities](https://www.cvedetails.com/vulnerability-list/vendor_id-19727/Materializecss.html), still present in materialize-css v1.0.

Honestly, since the declarations were almost the same, I had been using "package alias" so as to get TS support with this package, via `npm i -S @types/materializecss__materialize@npm:@types/materialize-css`. However, I decided to update the declarations, since new features and security changes were applied (making some properties to become deprecated) in [v1.1.0](https://github.com/materializecss/materialize/releases/tag/1.1.0-alpha).

File: types/materializecss__materialize/package.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "private": true,
+    "dependencies": {
+        "cash-dom": "*"
```
Comment (peterblazejewicz) [inline types/materializecss__materialize/package.json:4]:
```suggestion
        "cash-dom": "*"
```
this is no-n-, this should not be pass through versions, please use same from package or pick current ono (recall recent versioning dramas on npm)

File: types/materializecss__materialize/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+// Type definitions for materializecss__materialize 1.1
+// Project: https://materializecss.github.io/materialize/
+// Definitions by:  胡玮文 <https://github.com/huww98>
+//                  Maxim Balaganskiy <https://github.com/MaximBalaganskiy>
+//                  David Moniz <https://github.com/MonizDave>
+//                  Daniel Hoenes <https://github.com/broccoliarchy>
+//                  Rick Lucassen <https://github.com/SnowyLeopard>
+//                  Mauro Mascarenhas <https://github.com/mauromascarenhas/>
```
Comment (peterblazejewicz) [inline types/materializecss__materialize/index.d.ts:8]:
```ts
// Definitions by:  胡玮文 <https://github.com/huww98>
//                  Maxim Balaganskiy <https://github.com/MaximBalaganskiy>
//                  David Moniz <https://github.com/MonizDave>
//                  Daniel Hoenes <https://github.com/broccoliarchy>
//                  Rick Lucassen <https://github.com/SnowyLeopard>
//                  Mauro Mascarenhas <https://github.com/mauromascarenhas/>
```
I don't think this is ok to just copy over peoples handlers here without at least shout out in the PR asking for ACK (or expressing some kind of support in PR comments, etc)

File: types/materializecss__materialize/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.1
```
Comment (peterblazejewicz) [inline types/materializecss__materialize/index.d.ts:10]:
```suggestion
```
TS3.8 is the minimum version enforced here. If you have some specific TS versioning requirements at runtime (when DT are used), please use TS config paths versining (I assume that was just a copy over)

File: types/materializecss__materialize/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
+// Type definitions for materializecss__materialize 1.1
+// Project: https://materializecss.github.io/materialize/
```
Comment (peterblazejewicz) [inline types/materializecss__materialize/index.d.ts:2]:
```suggestion
// Project: https://materializecss.github.io/materialize/
```
so if this is a new definition for a fork, the project here is obviously wrong and can trick people

Comment (peterblazejewicz) [reply types/materializecss__materialize/index.d.ts:2]: skip this one, I can't read in the dark :)

File: types/materializecss__materialize/index.d.ts (hunk 2 above)
Comment (mauromascarenhas) [reply types/materializecss__materialize/index.d.ts]: Ok, just removed them.

File: types/materializecss__materialize/package.json (hunk 1 above)
Comment (mauromascarenhas) [reply types/materializecss__materialize/package.json]: Added major version requirement in last commit mauromascarenhas@ec93c3b.

File: types/materializecss__materialize/index.d.ts (hunk 3 above)
Comment (mauromascarenhas) [reply types/materializecss__materialize/index.d.ts]: Removed TS version requirement in last commit mauromascarenhas@ec93c3b.

**Comment by peterblazejewicz** [comment]
> was missing the following properties:
> 
> version
> keys
> tabPressed
> keyDown
> initializeJqueryWrapper

I don't think those are part of public api, so that's ok

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@mauromascarenhas thx!

**Comment by mauromascarenhas** [comment]
Ready to merge

**Pull request merged**